### ISSUE TITLE
How to hide the AvatarView with v1.0.0

### ISSUE THREAD

**Issue by janweinkauff** [MessageKit/MessageKit#674]
I'm trying to hide the `AvatarView` with the new changes of 1.0.0-beta.1. 

Right now I use `avatarView.isHidden = true` in `MessagesDisplayDelegate.configureAvatarView` which only hides the view but will not remove the space of the `AvatarView` due to the size from `outgoingAvatarSize` or `incomingAvatarSize` of `MessageSizeCalculator`.

Is it possible to change the value of `outgoingAvatarSize` or `incomingAvatarSize`? 

I couldn't find any delegate callback or entry point for a subclass.

**Comment by janweinkauff** [comment]
After a short chat in the slack channel I found a solution based on the suggestion from @creyD. 

`outgoingAvatarSize` or `incomingAvatarSize` are just properties of the `MessagesCollectionViewFlowLayout `.

```
if let layout = messagesCollectionView.collectionViewLayout as? MessagesCollectionViewFlowLayout {
  layout.textMessageSizeCalculator.outgoingAvatarSize = .zero
  ...
}
```

**Comment by Faisal0sal** [comment]
```
if let layout = messagesCollectionView.collectionViewLayout as? MessagesCollectionViewFlowLayout {
layout.textMessageSizeCalculator.outgoingAvatarSize = .zero
layout.textMessageSizeCalculator.incomingAvatarSize = .zero
}
```

doing this in viewdidload ^^ still gave me an avatar on outgoing messages

**Comment by janweinkauff** [comment]
I also did this change in the `viewDidLoad` method. It works on my side.
* Did you check if the cast `messagesCollectionView.collectionViewLayout as? MessagesCollectionViewFlowLayout` succeeds?
* Maybe set a breakpoint to `MessageSizeCalculator.avatarSize(for message:)` to understand from what change the `incomingAvatarSize` is set?

**Comment by Faisal0sal** [comment]
it works now, just moving it to the bottom of my viewDidLoad did it. not sure how this affect it.

**Comment by Faisal0sal** [comment]
Actually although this does work on text messages, once you start loading more messages it starts to flounder between having it and removing it. It also appears on media messages.

![simulator screen shot - iphone x - 2018-05-05 at 11 41 09](https://user-images.githubusercontent.com/10188948/39661407-bbbdf44e-5059-11e8-9fc7-419842814ba5.png)

**Comment by janweinkauff** [comment]
You can also set the avatar size for all other kind of messages. Please have a look into `MessagesCollectionViewFlowLayout`.

To change the avatar size you do same as desribed but with a different `MessageSizeCalculator`:
```
attributedTextMessageSizeCalculator.outgoingAvatarSize = .zero
emojiMessageSizeCalculator.outgoingAvatarSize = .zero
photoMessageSizeCalculator.outgoingAvatarSize = .zero
videoMessageSizeCalculator.outgoingAvatarSize = .zero
locationMessageSizeCalculator.outgoingAvatarSize = .zero
```

**Comment by Faisal0sal** [comment]
It still flounders 
![simulator screen shot - iphone x - 2018-05-05 at 13 43 26](https://user-images.githubusercontent.com/10188948/39662453-579f6df6-506a-11e8-986d-6f63609a5351.png)

**Comment by SD10** [comment]
@Faisal0sal keep in mind cells are reused

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ljs19923** [comment]
Do you know why i get this error ?
![image](https://user-images.githubusercontent.com/5727223/39705993-fbf832ec-51dd-11e8-82e8-cafc2d86975c.png)

**Comment by SD10** [comment]
@JulienLevallois I'm not sure exactly. What is your MessageKit version? Have you tried clearing derived data folder?

**Comment by ljs19923** [comment]
My version is 0.13.0 and yes i deleted this folder
![image](https://user-images.githubusercontent.com/5727223/39706373-38ae6746-51df-11e8-9299-f8f46ddf39ed.png)

**Comment by ljs19923** [comment]
I need to update to 0.13.5 or 1.0.0-beta.1 ?

Because with   pod 'MessageKit' and pod install i got this version 0.13.0

@SD10

**Comment by SD10** [comment]
You need `1.0.0-beta.1`
Run `pod repo update`, then `pod install`

**Comment by ljs19923** [comment]
I got this error :/

[!] Failed to connect to GitHub to update the CocoaPods/Specs specs repo - Please check if you are offline, or that GitHub is down

**Comment by Faisal0sal** [comment]
`pod 'MessageKit', '~> 1.0.0-beta.1'`

 This is what worked for me.

**Comment by ljs19923** [comment]
How can i verticaly align the avatar bottom with the cell in this new version ? 1.0.0-Beta.1

![image](https://user-images.githubusercontent.com/5727223/39713463-d87ce51c-51f4-11e8-82df-994a35ff0289.png)

**Comment by cwalo** [comment]
@JulienLevallois Please create a separate issue for your question. Thanks!

**Comment by ljs19923** [comment]
When i send a new message the outgoingAvatarSize = .zero didn't works

![image](https://user-images.githubusercontent.com/5727223/39713940-845fa436-51f6-11e8-8a3b-0b9c19610a09.png)

**Comment by jaunesarmiento** [comment]
Tried setting the avatar size using the following but I'm still getting blank spaces in the avatar area.

```swift
override func viewDidLoad() {
  super.viewDidLoad()

  // some code here

  if let layout = messagesCollectionView.collectionViewLayout as? MessagesCollectionViewFlowLayout {
    layout.textMessageSizeCalculator.outgoingAvatarSize = .zero
    layout.textMessageSizeCalculator.incomingAvatarSize = .zero
  }
}
```

![simulator screen shot - iphone 8 plus - 2018-05-09 at 13 04 54](https://user-images.githubusercontent.com/1166928/39796712-cb138b0a-538a-11e8-8a92-5f3f8fd9d1da.png)


What am I missing?

**Comment by SD10** [comment]
@jaunesarmiento because each case of `MessageKind` enum has its own `CellSizeCalculator` object as a property of `MessagesCollectionViewFlowLayout`. Look at the other properties in the class and set those to zero as well if you'd like

**Comment by jaunesarmiento** [comment]
Ah! @SD10 silly me. Thanks for pointing it out.

**Comment by ndonald2** [comment]
@SD10 it would be nice to somehow be able to set avatar sizes for all calculators at once rather than having to set size on each calculator individually, for convenience's sake. Do you think there's a place for that in the 1.0 API?

**Comment by SD10** [comment]
@ndonald2 Maybe 🤔 The maintenance burden for that would be lightweight but I dislike expanding the API surface area with something a user could do themselves. We would have to also look at a convenience method for the other calculator properties as well.

**Comment by ljs19923** [comment]
![image](https://user-images.githubusercontent.com/5727223/39873000-c4719910-5437-11e8-8eed-01122ee291c2.png)

I removed Avater view but how can i align my bottom date to the right ? without avatar.
Thanks

**Comment by ndonald2** [comment]
@SD10 
> I dislike expanding the API surface area with something a user could do themselves

I totally agree, but as you mentioned the maintenance cost for this would be quite low. In my opinion it seems like a very common use case to just want to change MK's default avatar size, which for both outgoing and incoming messages across all message types is currently _extremely_ verbose:

```swift
layout.textMessageSizeCalculator.incomingAvatarSize = avatarSize
layout.textMessageSizeCalculator.outgoingAvatarSize = avatarSize
layout.attributedTextMessageSizeCalculator.incomingAvatarSize = avatarSize
layout.attributedTextMessageSizeCalculator.outgoingAvatarSize = avatarSize
layout.videoMessageSizeCalculator.incomingAvatarSize = avatarSize
layout.videoMessageSizeCalculator.outgoingAvatarSize = avatarSize
layout.photoMessageSizeCalculator.incomingAvatarSize = avatarSize
layout.photoMessageSizeCalculator.outgoingAvatarSize = avatarSize
// ... etc for all message types
```

That's not to say that users couldn't just create an extension to do this, but if the maintenance cost is indeed low, why not just add to the API for a better dev experience?

**Comment by ljs19923** [comment]
@ndonald2 do you how can i am align the bottom date to the right ? Without avatars

**Comment by SD10** [comment]
@JulienLevallois please keep all discussions on topic

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ljs19923** [comment]
Its on the topic.. its about the avatar view

**Comment by nathantannar4** [comment]
@JulienLevallois Take a look at `func cellBottomLabelAlignment(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> LabelAlignment `. This is shown in the example project, please take a good look at it as I think it would help for a lot of your questions.

**Comment by Artheyn** [comment]
@JulienLevallois @SD10 , I've faced the same issues exposed by Julien and I've pushed a PR to fix this: https://github.com/MessageKit/MessageKit/pull/699

**Comment by commando24** [comment]
Now your can use 
`func avatarSize(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> CGSize {
    return .zero
  }`
to hide avatar.

**Comment by Artheyn** [comment]
Thanks for your reply @commando24 . In fact, this method has been deprecated since MessageKit 1.0:

```
    func avatarSize(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> CGSize {
        fatalError("avatarSize(for:at:in) has been removed in MessageKit 1.0.")
    }
```

But if you use the latest beta `1.0.0-beta.2` the issue is fixed (by https://github.com/MessageKit/MessageKit/pull/716).

You can now directly use : 

```
private func setupCollectionView() {
    guard let flowLayout = messagesCollectionView.collectionViewLayout as? MessagesCollectionViewFlowLayout else { return }

    flowLayout.attributedTextMessageSizeCalculator.outgoingAvatarSize = .zero
    flowLayout.attributedTextMessageSizeCalculator.incomingAvatarSize = .zero
}
```

**Comment by n13** [comment]
I should probably have checked this thread first. I was implementing the old delegate methods, and noticed they never get called. Duh. Ok. Then eventually found what @Artheyn said above.

```
guard let flowLayout = messagesCollectionView.collectionViewLayout as? MessagesCollectionViewFlowLayout else { return }

flowLayout.attributedTextMessageSizeCalculator.outgoingAvatarSize = .zero
flowLayout.attributedTextMessageSizeCalculator.incomingAvatarSize = .zero
```

**Comment by matsotaa** [comment]
Hey! nice pod)
I've got a problem, when I hide avatars in configureAvatarView:
when I type there
`if let layout = messagesCollectionView.collectionViewLayout as? MessagesCollectionViewFlowLayout {
        layout.textMessageSizeCalculator.outgoingAvatarSize = .zero
        layout.textMessageSizeCalculator.incomingAvatarSize = .zero
        }`

and if in func `messageReceived` I typed `messagesCollectionView.scrollToBottom(animated: true)` then.. in the moment when I entered in any chat, which have more messages than screen able to show scrollToBottom performs, but after it scrolls back to top.
How can I stop this scrolling(to top)
